src/api/app/models/status/check.rb

```diff
@@ -12,6 +12,7 @@ class Status::Check < ApplicationRecord
 
   #### Associations macros (Belongs to, Has one, Has many)
   belongs_to :checkable, polymorphic: true
+  belongs_to :repository_publish, -> { where(status_checks: { checkable_type: 'Status::RepositoryPublish' }).includes(:checks) }, foreign_key: 'checkable_id'
```

Comment:
If you go from `Repository` up there you can drop this too right?

Reply:
Dropped